doc/code/templates.rst

```diff
@@ -0,0 +1,134 @@
+.. _templates:
+
+Templates
+=========
+
+**Module name:** :mod:`pennylane.templates`
+
+.. currentmodule:: pennylane.templates
+
+This module provides a growing library of templates of common quantum
+machine learning circuit architectures that can be used to easily build,
+evaluate, and train more complex quantum machine learning models. In the
+quantum machine learning literature, such architectures are commonly known as an
+**ansatz**.
+
+PennyLane conceptually distinguishes two types of templates, **layer architectures** and **input embeddings**:
+
+* Layer architectures, found in :mod:`pennylane.templates.layers`, define blocks of gates that are repeated like the layers in a neural network. They usually contain only trainable parameters.
+
+* Embeddings, found in :mod:`pennylane.templates.embeddings`, encode input features into the quantum state of the circuit. These embeddings can also depend on trainable parameters, in which case the embedding is learnable.
+
+Each trainable template has a dedicated function in :mod:`pennylane.templates.parameters` which generates a list of **randomly initialized** arrays for the trainable **parameters**. The entries of the list contain valid positional arguments for the template, allowing for the syntax ``MyTemplate(*par_list)``.
+
+.. note::
+
+    Templates are constructed out of **structured combinations**
+    of the :mod:`quantum operations <pennylane.ops>` provided by PennyLane.
+    As a result, you should follow all the rules of quantum operations
+    when you use templates. For example **template functions can only be
+    used within a valid** :mod:`pennylane.qnode`.
+
+
+Summary
+-------
+
+.. rst-class:: contents local topic
+
+.. toctree::
+    :maxdepth: 3
+
+    templates/layers
+    templates/embeddings
+    templates/parameters
+
+Examples
+--------
+
+You can construct a circuit-centric quantum
+classifier with the architecture from :cite:`schuld2018circuit` on an arbitrary
+number of wires and with an arbitrary number of blocks by using the
+template :func:`~.StronglyEntanglingLayers` in the following way:
+
+.. code-block:: python
+
+	import pennylane as qml
+	from pennylane.templates.layers import StronglyEntanglingLayers
+	from pennylane.templates.parameters import parameters_stronglyentangling_layers
+
+	from pennylane import numpy as np
+
+	num_wires = 4
+	num_blocks = 3
+
+	dev = qml.device('default.qubit', wires=num_wires)
+
+
+	@qml.qnode(dev)
+	def circuit(pars, x=None):
+	    qml.BasisState(x, wires=range(num_wires))
+	    StronglyEntanglingLayers(*pars, periodic=True, wires=range(num_wires))
```

Comment:
the parameter 'periodic=True' is missing in the function definition in the layers files.

Reply:
That was actually a mistake, I took it out to use periodic boundary conditions by default (in the interest of user simplicity). The example code is running now.